### PR TITLE
[GEOT-7404] Add SQLite Connection Timeout in case of hung GeoPackage connections

### PR DESCRIPTION
**Pull request by turingtestfail**
[GEOT-7404](https://osgeo-org.atlassian.net/browse/GEOT-7404) [Powered by Pull Request Badge](https://pullrequestbadge.com/?utm_medium=github&utm_source=geotools&utm_campaign=badge_info)

# Checklist

- [x] I have read the [contribution guidelines](https://github.com/geotools/geotools/blob/main/CONTRIBUTING.md).
- [x] I have sent a [Contribution Licence Agreement](https://docs.geotools.org/latest/developer/procedures/contribution_license.html) (not required for small changes, e.g., fixing typos in documentation).
- [x] First PR targets the `main` branch (backports managed later; ignore for branch specific issues).
- [x] Avoid [Java 9+ split packages](http://tutorials.jenkov.com/java/modules.html#split-packages-not-allowed).
- [ ] All the build checks are green ([see automated QA checks](https://docs.geotools.org/latest/developer/conventions/code/qa.html)).

For core and extension modules:

- [x] New unit tests have been added covering the changes.
- [ ] [Documentation](https://github.com/geotools/geotools/tree/main/docs) has been updated (if change is visible to end users).
- [x] There is an issue in [GeoTools Jira](https://osgeo-org.atlassian.net/projects/GEOT) (except for changes not visible to end users). 
- [x] Commit message(s) must be in the form ``[GEOT-XYZW] Title of the Jira ticket``.
- [x] Bug fixes and small new features are presented as a single commit.
- [ ] The commit targets a single objective (if multiple focuses cannot be avoided, each one is in its own commit, and has a separate ticket describing it).

### CODE REVIEW

File: modules/plugin/geopkg/src/test/java/org/geotools/geopkg/GeoPkgDataStoreFactoryTest.java [hunk 1]
```diff
@@ -54,7 +137,6 @@ public void testBaseDirectory() throws IOException {
         DataStore data = factory.createDataStore(map);
         data.createSchema(b.buildFeatureType());
         data.dispose();
-
-        assertTrue(new File(tmp.getRoot(), "foo.gpkg").exists());
+        return data;
```
Comment (aaime) [inline modules/plugin/geopkg/src/test/java/org/geotools/geopkg/GeoPkgDataStoreFactoryTest.java:140]: "data" here is dead, it has been disposed of. It's also not used anywhere I can see.
So I'd switch the method to void.

**Review by aaime — CHANGES_REQUESTED** [review]
I would rename the ticket title, this is just a GeoPackage change, even if we noticed the trouble in the context of vector mosaic. See also the comment below.

File: modules/plugin/geopkg/src/test/java/org/geotools/geopkg/GeoPkgDataStoreFactoryTest.java (hunk 1 above)
Comment (turingtestfail) [reply modules/plugin/geopkg/src/test/java/org/geotools/geopkg/GeoPkgDataStoreFactoryTest.java]: fixed

**Comment by turingtestfail** [comment]
> I would rename the ticket title, this is just a GeoPackage change, even if we noticed the trouble in the context of vector mosaic. See also the comment below.

Fixed

**Review by aaime — APPROVED** [review]

**Pull request merged**